Code/Tools/ProjectManager/Platform/Mac/O3DE_SDK_Launcher.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzFramework/Process/ProcessWatcher.h>
+
+#include <cstdlib>
+#include <mach-o/dyld.h>
+
+int main(int argc, char* argv[])
+{
+    // We need to pass in the engine path since we won't be able to find it by searching upwards.
+    // We can't use any containers that use our custom allocator till after the call to ComponentApplication::Create()
+    AZ::IO::FixedMaxPath processPath = AZ::Utils::GetExecutableDirectory();
+    AZ::IO::FixedMaxPath enginePath = (processPath / "../Engine").LexicallyNormal();
+    auto enginePathParam = AZ::SettingsRegistryInterface::FixedValueString::format(R"(--engine-path="%s")", enginePath.c_str());
+    // Uses the fixed_vector deduction guide to determine the type is AZStd::fixed_vector<char*, 3>
+    AZStd::fixed_vector commandLineParams{ processPath.Native().data(), enginePathParam.data(), static_cast<char*>(nullptr) };
+
+
+    
+    // Create a ComponentApplication to initialize the AZ::SystemAllocator and initialize the SettingsRegistry
+    AZ::ComponentApplication application(2, commandLineParams.data());
+    application.Create(AZ::ComponentApplication::Descriptor());
+ 
+    AZ::IO::FixedMaxPath installedBinariesFolder;
+    if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+    {
+        if (settingsRegistry->Get(installedBinariesFolder.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_InstalledBinaryFolder))
+        {
+            installedBinariesFolder = enginePath / installedBinariesFolder;
+        }
+    }
+
+    AZ::IO::FixedMaxPath shellPath = "/bin/sh";
+    AZStd::string parameters = AZStd::string::format("-c \"export LY_CMAKE_PATH=/usr/local/bin && \"%s/python/get_python.sh\"\"", enginePath.c_str());
+    AzFramework::ProcessLauncher::ProcessLaunchInfo shellProcessLaunch;
+    shellProcessLaunch.m_processExecutableString = AZStd::move(shellPath.Native());
+    shellProcessLaunch.m_commandlineParameters = parameters;
+    shellProcessLaunch.m_showWindow = true;
+    shellProcessLaunch.m_workingDirectory = AZStd::string::format("%s", enginePath.c_str());
```

Comment:
The Path classes has a `String()` function to return a new copy of the path as an `AZStd::string`.
There is a similiar function for making a copy of the path and returning a `AZStd::fixed_string` called `FixedMaxPathString()`

There this line can be replaced with.
```suggestion
    shellProcessLaunch.m_workingDirectory = enginePath.Native();
```

Reply:
Will fix